Marlin/Marlin_main.cpp

```diff
@@ -9312,6 +9329,318 @@ inline void gcode_M355() {
+  }
+
+  //Checks and reports the status of a given encoder module
+  inline void gcode_M861() {
+    bool reported = false;
+    LOOP_XYZ(i) {
+      if (code_seen(axis_codes[i])) {
+        i2cEncoderManager.report_status(AxisEnum(i));
+        reported = true;
+      }
+    }    
+    
+    if(!reported) {
+      LOOP_XYZ(i) {
+        SERIAL_ECHO(axis_codes[i]);
+        SERIAL_ECHO(": ");
+        i2cEncoderManager.report_status(AxisEnum(i));
+      }
+    }
+  }
+
+  //Performs an axis continuity test for a given encoder module / axis
+  inline void gcode_M862() {
+    AxisEnum selectedAxis;
+    bool axisSelected = false;
+
+    for(int i = 0; i < NUM_AXIS; i++) {
+      if (code_seen(axis_codes[i])) {
+        selectedAxis = AxisEnum(i);
+        axisSelected = true;
+      }
+    }
+
+    if(axisSelected) {
+      i2cEncoderManager.test_axis(selectedAxis);
+    } else {
+      i2cEncoderManager.test_axis();
+    }
+  }
+
+  //Performs an automatic steps per mm calibration for a given encoder module / axis
+  inline void gcode_M863() {
+    AxisEnum selectedAxis;
+    int iterations = 1;
+    bool axisSelected = false;
+
+    for(int i = 0; i < NUM_AXIS; i++) {
+      if (code_seen(axis_codes[i])) {
+        selectedAxis = AxisEnum(i);
+        axisSelected = true;
+      }
+    }
+
+    if (code_seen('I')) {
+      iterations = constrain(code_value_int(),1,10);
+    }
+
+    if(axisSelected) {
+      i2cEncoderManager.calibrate_steps_mm(selectedAxis,iterations);
+    } else {
+      i2cEncoderManager.calibrate_steps_mm(iterations);
+    }
+
+  }
+/*
+  //Changes a module from one I2C address to another
+  // assumes module is currently at default address,
+  // specify different current address by OXXX (i.e. O032)
+  // either specify X/Y/Z/E to select new address, or AXXX (i.e. A031)
+  inline void gcode_M864() {
+    AxisEnum selectedAxis;
+    int newAddress, oldAddress;
+    bool addressSelected = false;
+
+    for(int i = 0; i < NUM_AXIS; i++) {
+      if (code_seen(axis_codes[i])) {
+        selectedAxis = AxisEnum(i);
+        addressSelected = true;
+      }
+    }
+
+    if (addressSelected == false) {
+      if(code_seen('A')) {
+        newAddress = code_value_int();
+        addressSelected = true;
+      }
+    } else {
+      switch (selectedAxis) {
+        case X_AXIS:
+          newAddress = I2C_ENCODER_PRESET_ADDR_X;
+          break;
+        case Y_AXIS:
+          newAddress = I2C_ENCODER_PRESET_ADDR_Y;
+          break;
+        case Z_AXIS:
+          newAddress = I2C_ENCODER_PRESET_ADDR_Z;
+          break;
+        case E_AXIS:
+          newAddress = I2C_ENCODER_PRESET_ADDR_E;
+          break;
+      }
+    }
+
+    if(code_seen('O')) {
+      oldAddress = code_value_int();
+    } else {
+      oldAddress = I2C_ENCODER_PRESET_ADDR_X;
+    }
+
+    if(addressSelected) { 
+      if(newAddress > 0 && newAddress <= 255 && oldAddress > 0 && oldAddress <= 255) {
+        SERIAL_ECHO("Changing module at address ");
```

Comment:
Yes.  Variables are coming directly from SRAM, so you can't use the PGM/progmem functions, which pull from the flash.